android/ui-android/src/main/java/com/trafi/ui/component/Button.kt

```diff
@@ -20,40 +42,138 @@ import com.trafi.ui.theme.lightColors
-        modifier = modifier.heightIn(min = constants.height).fillMaxWidth(),
         enabled = enabled,
         shape = if (constants.cornerRadius.isRound) {
             RoundedCornerShape(percent = 50)
         } else {
             RoundedCornerShape(constants.cornerRadius)
         },
         colors = androidx.compose.material.ButtonDefaults.buttonColors(
-            backgroundColor = color,
-            disabledBackgroundColor = disabledColor,
-            contentColor = contentColor,
-            disabledContentColor = disabledContentColor
-        )
+            backgroundColor = backgroundColor.takeOrElse { constants.defaultBackgroundColor },
+            disabledBackgroundColor = backgroundColor.takeOrElse { constants.disabledBackgroundColor },
+            contentColor = contentColor.takeOrElse { constants.defaultContentColor },
+            disabledContentColor = contentColor.takeOrElse { constants.disabledContentColor },
+        ),
+        elevation = androidx.compose.material.ButtonDefaults.elevation(
+            defaultElevation = elevation ?: 2.dp,
+            pressedElevation = elevation ?: 8.dp,
+        ),
+        contentPadding = PaddingValues(
+            start = constants.paddingHorizontal,
+            top = if (small) constants.paddingVerticalSmall else constants.paddingVertical,
+            end = constants.paddingHorizontal,
+            bottom = if (small) constants.paddingVerticalSmall else constants.paddingVertical,
+        ),
+        modifier = modifier
+            .defaultMinSizeConstraints(minHeight = if (small) constants.heightSmall else constants.height)
+            .run { if (wrapped) this else fillMaxWidth() }
     ) {
+        @Suppress("NAME_SHADOWING")
+        val contentColor = contentColor.takeOrElse { constants.contentColor(enabled) }
+        val iconModifier = Modifier
+            .padding(end = constants.spaceBetween)
+            .size(if (small) constants.iconSizeSmall else constants.iconSize)
+            .align(Alignment.CenterVertically)
+        AnimatedVisibility(loading || icon != null) {
+            if (loading) {
+                CircularProgressIndicator(
+                    color = contentColor,
+                    strokeWidth = 1.5.dp,
+                    modifier = iconModifier,
+                )
+            } else icon?.let {
+                Icon(
+                    painter = icon,
+                    contentDescription = null,
+                    tint = contentColor,
+                    modifier = iconModifier,
+                )
+            }
+        }
         Text(
             text,
-            style = constants.textStyle
+            color = contentColor,
+            style = if (small) constants.textStyleSmall else constants.textStyle,
         )
     }
 }
 
+@Composable
+private fun ButtonConstants.contentColor(enabled: Boolean): Color =
+    if (enabled) defaultContentColor else disabledContentColor
+
+@Composable
+public fun SecondaryButton(
```

Comment:
It seems to be in line with other Compose components (e.g. Material components) to declare common variants as separate components that delegate to the more generic component (as opposed to e.g. using an enum). I've added `SecondaryButton` and `TertiaryButton` with that in mind.

cc @mikasdo @Ignelios

Reply:
Nice i think we should implement this too.